Common/Serialize/Serializer.h

```diff
@@ -120,7 +139,13 @@ class PointerWrap
 	void DoMarker(const char *prevName, u32 arbitraryNumber = 0x42);
 
 private:
+	size_t Offset() const { return *ptr - ptrStart_; }
+
 	const char *firstBadSectionTitle_ = nullptr;
+	u8 *ptrStart_;
+	std::vector<SerializeCheckpoint> checkpoints_;
```

Comment:
Not sure if it matters much, but might be nice to `reserve()` this to some value near its typical size?  Though, this probably is quick compared to all the memory copying save state writing does, anyway.

Would be nice to rebase out the accidental pspautotests revert.

-[Unknown]

Reply:
True, it's worth doing a reserve here, I'll check about how many sections we have. And I'll obviously revert the revert.